src/Parsers/ParserCreateQuery.cpp

```diff
@@ -1560,15 +1562,33 @@ bool ParserCreateViewQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expec
         if (!table_name_p.parse(pos, to_table, expected))
             return false;
     }
+    bool aliases_override = false;
 
     /// Optional - a list of columns can be specified. It must fully comply with SELECT.
-    if (s_lparen.ignore(pos, expected))
+    if (s_lparen.ignore(pos, expected)) // Parsing cases like CREATE VIEW (a Int64, b Int64) (a, b) SELECT ...
```

Comment:
So this is implemented in `ParserCreateViewQuery::parseImpl` which, according to l. 553 in src/Parsers/ParserCreateQuery.h, runs not only for `CREATE VIEW` but also for `CREATE TABLE` etc. Is this what we really want or should we restrict the code to views?

Reply:
it is the `ParserCreateViewQuery` class, on l. 622, which is responsible for creating views only